src/network/stream_reader.rs

```diff
@@ -21,33 +21,31 @@
 //!
 
 use std::fmt;
-use std::io;
-use std::io::Read;
+use std::io::{self, BufRead};
 
-use network::message::RawNetworkMessage;
-use consensus::encode;
+use consensus::{encode, Decodable};
 
 /// Struct used to configure stream reader function
-pub struct StreamReader<R> {
+pub struct StreamReader<R: BufRead> {
```

Comment:
When you use `BufRead` you need to wrap something like a file in a `BufReader`.

Why would we want this overhead on the users?

Reply:
Because I'd like to coner different types of inputs. For instance, with plain `BufReader` I was not able to read from Stdin. You can see a sample here: https://github.com/pandoracore/bpdump/blob/73ae835f223f8ad5b93aef6daf2bbd7e60901722/src/main.rs#L38 This line would not compile if there were just `BufReader`.

More explanations can be found here: https://stackoverflow.com/questions/36088116/how-to-do-polymorphic-io-from-either-a-file-or-stdin-in-rust

Reply:
Is there anywhere that we use the features of `BufRead` that aren't available in `Read`? Otherwise this seems unnecessarily restrictive.

Reply:
Just re-checked, there is no. Agree, correct, I'll replace it with `Read`